include/swift/Reflection/ReflectionContext.h

```diff
@@ -644,6 +645,46 @@ class ReflectionContext
     return false;
   }
 
+  /// Adds an image using the FindSection closure to find the swift metadata
+  /// sections. \param[in] FindSection
+  ///     Closure that finds sections by name. ReflectionContext is in charge
+  ///     of freeing the memory buffer in the RemoteRef return value.
+  ///     process.
+  /// \return
+  ///     /b True if any of the reflection sections were registered,
+  ///     /b false otherwise.
+  bool
+  addImage(std::function<std::pair<RemoteRef<void>, uint64_t>(llvm::StringRef)>
```

Comment:
this could be a `llvm::function_ref` to indicate it's a non-escaping function/lambda.